microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,278 @@
+  const url = new URL(tab.url);
+  // Enables the side panel on bing.com
+  if (url.origin === BING_ORIGIN) {
+    await chrome.sidePanel.setOptions({
+      tabId,
+      path: 'sidepanel.html',
+      enabled: true
+    });
+  } else {
+    // Disables the side panel on all other sites
+    await chrome.sidePanel.setOptions({
+      tabId,
+      enabled: false
+    });
+  }
+});
+```
+
+When a user switches to a tab where the sidebar is not enabled, the sidebar will be hidden.
+
+
+<!-- ------------------------------ -->
+#### Example 4: Enabling the action icon to open the sidebar
+
+By declaring the `action` key in the manifest, you can allow users to open the side panel when they click on the action toolbar icon with `sidePanel.setPanelBehavior()`.
+
+`manifest.json`:
+
+```json
+{
+  "name": "My sidebar extension",
+  ...
+   "action": {
+    "default_title": "Click to open panel"
+  },
+  ...
+}
+```
+
+`service-worker.js`:
+
+```js
+const BING_ORIGIN = 'https://www.bing.com';
+
+// Allows users to open the side panel by clicking on the action toolbar icon
+chrome.sidePanel
+  .setPanelBehavior({ openPanelOnActionClick: true })
+  .catch((error) => console.error(error));
+...
+```
+
+
+<!-- ====================================================================== -->
+## Sidebar user experience
+
+
+<!-- ------------------------------ -->
+#### Sidebar icon next to the extension's name
+
+* The sidebar icon is displayed next to the extension name in the hub in the toolbar:
+
+![Sidebar dialog](./sidebar-images/sidebar-dialog.png)
+<!-- todo: put real content in png -->
+
+* On the Microsoft Edge Add-Ons store page, there's a dialog on the extension product page, highlighting the sidebar feature.
+<!-- todo: show that "dialog", is it actually a dialog? -->
+
+
+<!-- ------------------------------ -->
+#### Opening a extension in the sidebar
+
+To open an extension in the sidebar, the user can do either of the following:
+
+* Click the sidebar icon in the hub in the toolbar.
+
+* Right-click the extension's shortcut in the toolbar.
+
+
+<!-- ====================================================================== -->
+## Properties for sidebars
+
+
+<!-- ------------------------------ -->
+#### Panel behavior
+
+| Property | Description |
+|---|---|
+| `openPanelOnActionClick` | Whether clicking the extension's icon toggles showing the extension's entry in the sidebar.  The default value is `false`. |
+
+
+<!-- ------------------------------ -->
+#### Panel options
+
+| Property | Description |
+|---|---|
+| `tabId` | If specified, the sidebar options only apply to the tab that has this ID.  If not specified, these options set the default behavior enabled.  Whether the sidebar is enabled.  The default value is `true`.<!--todo: delete "Whether the sidebar is enabled.  The default value is `true`."? --> |
+| `path` | The path to the side panel HTML file to use.  This needs to be a local resource within the extension package. |
```

Comment:
```suggestion
| `path` | The path to the sidebar HTML file to use.  This needs to be a local resource within the extension package. |
```
definitions table at top probably allows preferring "sidebar" here, unless we want to say "panel" to match API name.

Reply:
fixed locally